shared/signing/evm/evm_transaction.go

```diff
@@ -101,25 +101,30 @@ func WithNonce(address string) TxBuildOption {
 	}
 }
 
-func WithEstimatedGasLimit(from, to, value string, data []byte) TxBuildOption {
-	return func(ctx context.Context, b TxBuilder, tx *types.DynamicFeeTx) error {
-		to := common.HexToAddress(to)
+// EstimateGasForTx estimates the gas needed for a transaction with the given parameters
+func (b TxBuilder) EstimateGasForTx(ctx context.Context, from, to, value string, data []byte) (uint64, error) {
+	toAddr := common.HexToAddress(to)
```

Comment:
i feel like the `TxBuilder` `WithEstimatedGasLimit` function should keep the node calling logic in it, so it is consistent with how all of the other functions work, like `WithNonce`, `WithEstimatedGasTipCap`, etc. I don't think its a big deal to make an extra node call to estimate the gas twice (once when building the actual tx that we are going to send, and once when checking if the estimated gas for the tx is > their configured max rebalance tx gas amount). I think if we want a separate function to do the gas estimation and return the amount, it should maybe be on a different struct, since it doesn't really make sense for the tx builder.

Reply:
makes sense, reverted to original setup